playwright/e2e/crypto/backups.spec.ts

```diff
@@ -0,0 +1,132 @@
+            .locator("..")
+            .click();
+
+        await expectBackupVersionToBe(page, "1");
+
+        await tab.getByRole("button", { name: "Delete Backup", exact: true }).click();
+
+        await expect(currentDialogLocator.getByRole("heading", { name: "Delete Backup" })).toBeVisible();
+
+        // Delete it
+        await currentDialogLocator.getByTestId("dialog-primary-button").click(); // Click "Delete Backup"
+
+        // Create another
+        await tab.getByRole("button", { name: "Set up", exact: true }).click();
+
+        await expect(currentDialogLocator.getByRole("heading", { name: "Security Key" })).toBeVisible();
+
+        await currentDialogLocator.getByLabel("Security Key").fill(securityKey);
+        await currentDialogLocator.getByRole("button", { name: "Continue", exact: true }).click();
+
+        await expect(currentDialogLocator.getByRole("heading", { name: "Success!" })).toBeVisible();
+
+        await currentDialogLocator.getByRole("button", { name: "OK", exact: true }).click();
+
+        // Open the settings again
+        await app.settings.openUserSettings("Security & Privacy");
+        await expect(tab.getByRole("heading", { name: "Secure Backup" })).toBeVisible();
+
+        // expand the advanced section to see the active version in the reports
+        await page
+            .locator(".mx_Dialog .mx_SettingsSubsection_content details .mx_SecureBackupPanel_advanced")
+            .locator("..")
+            .click();
+
+        await expectBackupVersionToBe(page, "2");
+
+        // ==
+        // Ensure that if you don't have the secret storage passphrase the backup won't be created
+        // ==
+
+        // First delete version 2
+        await tab.getByRole("button", { name: "Delete Backup", exact: true }).click();
+
+        await expect(currentDialogLocator.getByRole("heading", { name: "Delete Backup" })).toBeVisible();
+        await currentDialogLocator.getByTestId("dialog-primary-button").click(); // Click "Delete Backup"
+
+        // Try to create another
+        await tab.getByRole("button", { name: "Set up", exact: true }).click();
+
+        await expect(currentDialogLocator.getByRole("heading", { name: "Security Key" })).toBeVisible();
+
+        // But cancel the security key dialog, to simulate not having the secret storage passphrase
+        await currentDialogLocator.getByTestId("dialog-cancel-button").click();
+
+        await expect(currentDialogLocator.getByRole("heading", { name: "Starting backup…" })).toBeVisible();
+        // check that it failed
+        await expect(currentDialogLocator.getByText("Unable to create key backup")).toBeVisible();
+
+        // cancel
+        await currentDialogLocator.getByTestId("dialog-cancel-button").click();
+
+        // go back to the settings to check that no backup was created (the setup button should still be there)
+        const tab2 = await app.settings.openUserSettings("Security & Privacy");
```

Comment:
why do we create a new `tab2` here, but not the other two times we re-open the settings? Surely either we can re-use `tab` or we can't?